src/grammar.coffee

```diff
@@ -429,12 +436,7 @@ grammar =
     o 'Value OptFuncExist String',              -> new TaggedTemplateCall $1, $3, $2
     o 'Value OptFuncExist Arguments',           -> new Call $1, $3, $2
     o 'Invocation OptFuncExist Arguments',      -> new Call $1, $3, $2
-    o 'Super'
-  ]
-
-  Super: [
-    o 'SUPER',                                  -> new SuperCall
-    o 'SUPER Arguments',                        -> new SuperCall $2
+    o 'SUPER OptFuncExist Arguments',           -> new SuperCall LOC(1)(new Super), $3, $2
```

Comment:
Yeh, so `SUPER Arguments` has been directly subsumed by `SUPER OptFuncExists Arguments` (which allows `super?()` to parse).

The `SUPER` case has been dropped as, in light of supporting `super` with accessors, it becomes confusing whichever way we spin it:

- Keep bare `super` as a super call to the method of the same name, forwarding arguments. This is fine until we have to deal with `super.foo`, which now looks ambiguous: is it `super.foo` as it appears, or `super().foo`? What about `(super).foo`? Tbh this use of `super` is useful enough that I'd be happy to leave it in, but it does lead to a number of ways of doing the same thing with `super`.

- Allow bare `super`, and handle it the same way `super()` is handled (compiler error in a `constructor`, and otherwise a reference to `super.<current method>`). This would be safe and consistent, except it could lead to subtle bugs and misunderstandings for people upgrading to CS2 from CS1.

SO, a bare `super` will no longer parse. Unfortunately the [error message](https://github.com/jashkenas/coffeescript/pull/4424/files#diff-55324284aab67fc5ce7570a4f82c6336R1321) isn't very friendly, and any tips to improve that would be great!